src/ITILTemplateField.php

```diff
@@ -117,7 +120,171 @@ abstract public static function getExcludedFields();
+            if (is_subclass_of(static::class, ITILTemplatePredefinedField::class)) {
+                $embedded_ma_params = [
+                    'id_field'         => '__VALUE__',
+                    'itemtype'         => static::$itiltype,
+                    'inline'           => true,
+                    'submitname'       => _sx('button', 'Add'),
+                    'options'          => [
+                        'relative_dates'     => 1,
+                        'with_time'          => 1,
+                        'with_days'          => 0,
+                        'with_specific_date' => 0,
+                        'itemlink_as_string' => 1,
+                        'entity'             => $tt->getEntityID()
+                    ]
+                ];
+                $extra_form_html = Ajax::updateItemOnSelectEvent(
+                    "dropdown_num{$rand}",
+                    "show_massiveaction_field",
+                    $CFG_GLPI["root_doc"] . "/ajax/dropdownMassiveActionField.php",
+                    $embedded_ma_params,
+                    false
+                );
+                $extra_form_html .= "<div id='show_massiveaction_field'></div>";
+            }
+
+            $twig_params = [
+                'form_url' => static::getFormURL(),
+                'items_id_field' => static::$items_id,
+                'itemtype_name' => static::getTypeName(),
+                'used' => $used,
+                'id' => $ID,
+                'fields' => $fields_dropdown_values,
+                'extra_form_html' => $extra_form_html,
+                'rand' => $rand,
+                'show_submit' => !is_subclass_of(static::class, ITILTemplatePredefinedField::class)
+            ];
+            echo TemplateRenderer::getInstance()->renderFromStringTemplate(<<<TWIG
+                {% import 'components/form/fields_macros.html.twig' as fields %}
+                {% import 'components/form/basic_inputs_macros.html.twig' as inputs %}
+                <div>
+                    <form name="itiltemplatehidden_form{{ rand }}" method="post" action="{{ form_url }}" data-submit-once>
+                        {{ inputs.hidden('_glpi_csrf_token', csrf_token()) }}
+                        {{ inputs.hidden(items_id_field, id) }}
+                        <div class="d-flex justify-content-center flex-wrap">
+                            {{ fields.dropdownArrayField('num', 0, fields, null, {
+                                no_label: true,
+                                used: used,
+                                add_field_attribs: {
+                                    'aria-label': itemtype_name
+                                },
+                                rand: rand
+                            }) }}
+                            {{ extra_form_html|raw }}
+                            {% if show_submit %}
+                                <div class="ms-2">{{ inputs.submit('add', _x('button', 'Add')) }}</div>
```

Comment:
@cconard96 `_x()` will not be detected by `xgettext` here.

I suggest to move this template into a dedicated Twig file. You could, inside this file, include `components/datatable.html.twig`, in order to have a single call to `TemplateRenderer::getInstance()->display()` inside the current method.